shell/platform/tizen/tizen_event_loop.cc

```diff
@@ -87,6 +89,64 @@ void TizenEventLoop::ExcuteTaskEvents(void* data, void* buffer,
     }
     ecore_timer_add(flutter_duration, TaskTimerCallback, tizenEventLoop);
   } else {
-    tizenEventLoop->on_task_expired_(&(p_task->task));
+    {
+      std::lock_guard<std::mutex> lock(tizenEventLoop->expired_tasks_mutex_);
+      tizenEventLoop->expired_tasks_.push_back(p_task->task);
+    }
+    tizenEventLoop->OnTaskExpired();
+  }
+}
+
+TizenPlatformEventLoop::TizenPlatformEventLoop(
+    std::thread::id main_thread_id, TaskExpiredCallback on_task_expired)
+    : TizenEventLoop(main_thread_id, on_task_expired) {}
+
+TizenPlatformEventLoop::~TizenPlatformEventLoop() {}
+
+void TizenPlatformEventLoop::OnTaskExpired() {
+  for (const auto& task : expired_tasks_) {
+    on_task_expired_(&task);
+  }
+  expired_tasks_.clear();
```

Comment:
Shouldn't we lock expired_tasks_mutex_ here?

Reply:
I designed to lock the mutex in the caller of ```OnTaskExpired``` if necessary.

Reply:
@bbrto21 Apparently the caller (`ExcuteTaskEvents`) of `OnTaskExpired` doesn't hold the lock and the `TizenRenderEventLoop::OnTaskExpired()` implementation uses the lock. I'm not still sure what you intended.

Reply:
@swift-kim Well, now that I've reviewed it again.... I think that it looks good to lock the mutex here too. It's been so long since I implemented this part, so I can't remember exactly what I was thinking then.  :grin: :grin: :grin: